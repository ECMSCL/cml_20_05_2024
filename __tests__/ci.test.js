jest.mock('../src/dvc');

const { git } = require('../src/utils');
const report = require('../src/report');
const dvc = require('../src/dvc');
const ci = require('../src/ci');

describe('Dvc Report', () => {
  beforeEach(() => {
    report.same_warning = jest.fn(report.same_warning);

    git.log = jest.fn(git.log);
    git.revparse = jest.fn(refs => {
      return refs[0] === 'HEAD'
        ? '0000000000000000000000000000000000000000'
        : '1111111111111111111111111111111111111111';
    });
  });

  test("Paramters 'from' and 'to' are different, no warning will be displayed", async () => {
    const dvc_diff = {};
    const dvc_metrics_diff = {};
    const others = { all: [] };
    const from = 'HEAD~1';
    const to = 'HEAD';

    dvc.diff.mockResolvedValue(dvc_diff);
    dvc.metrics_diff.mockResolvedValue(dvc_metrics_diff);
    git.log.mockResolvedValue(others);

    await ci.dvc_report({ from, to });

    expect(git.revparse.mock.calls.length).toBe(2);
    expect(report.same_warning.mock.calls.length).toBe(0);
  });

  test('Comparing with itself, warning is displayed', async () => {
    jest.mock('../src/report');
    const dvc_diff = {};
    const dvc_metrics_diff = {};
    const others = { all: [] };
    const from = 'HEAD';
    const to = 'HEAD';

    dvc.diff.mockResolvedValue(dvc_diff);
    dvc.metrics_diff.mockResolvedValue(dvc_metrics_diff);
    git.log.mockResolvedValue(others);

    await ci.dvc_report({ from, to });
    expect(git.revparse.mock.calls.length).toBe(2);
    expect(report.same_warning.mock.calls.length).toBe(1);
  });
});
