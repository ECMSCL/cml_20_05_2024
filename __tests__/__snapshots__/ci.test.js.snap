// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dvc Report no data Different from to: No warning header 1`] = `
"### Baseline: HEAD~1 ( 1111111 vs 0000000 ) 
 

#### Metrics 

 No metrics available 

#### Data 

No metrics available 

#### Other experiments 
No other experiments available"
`;

exports[`Dvc Report no data Different from to: No warning header 2`] = `
"
<!doctype html>
<html>
	<head>
		<meta charset=\\"utf-8\\">
		<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, minimal-ui\\">
		<title>DVC Report</title>
		<link rel=\\"stylesheet\\" href=\\"report.css\\">
		<style>
			body {
				box-sizing: border-box;
				min-width: 200px;
				max-width: 980px;
				margin: 0 auto;
				padding: 45px;
			}
		</style>
	</head>
	<body>
      <div class=\\"markdown-body\\" id=\\"content\\">
        <h3 id=\\"baseline-head1--1111111-vs-0000000-\\">Baseline: HEAD~1 ( 1111111 vs 0000000 )</h3>
<h4 id=\\"metrics\\">Metrics</h4>
<p>No metrics available </p>
<h4 id=\\"data\\">Data</h4>
<p>No metrics available </p>
<h4 id=\\"other-experiments\\">Other experiments</h4>
<p>No other experiments available</p>
      </div>
  </body>
</html>
"
`;

exports[`Dvc Report no data Same from to: Displays warning header 1`] = `
"### Baseline: HEAD ( 0000000 vs 0000000 ) 
>:warning: You are comparing ref HEAD with itself, no diff available. 
Please [setup rev environment variable](https://github.com/iterative/dvc-cml#env-variables) accordingly 

#### Metrics 

 No metrics available 

#### Data 

No metrics available 

#### Other experiments 
No other experiments available"
`;

exports[`Dvc Report no data Same from to: Displays warning header 2`] = `
"
<!doctype html>
<html>
	<head>
		<meta charset=\\"utf-8\\">
		<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, minimal-ui\\">
		<title>DVC Report</title>
		<link rel=\\"stylesheet\\" href=\\"report.css\\">
		<style>
			body {
				box-sizing: border-box;
				min-width: 200px;
				max-width: 980px;
				margin: 0 auto;
				padding: 45px;
			}
		</style>
	</head>
	<body>
      <div class=\\"markdown-body\\" id=\\"content\\">
        <h3 id=\\"baseline-head--0000000-vs-0000000-\\">Baseline: HEAD ( 0000000 vs 0000000 )</h3>
<blockquote>
  <p>⚠️ You are comparing ref HEAD with itself, no diff available. <br />
  Please <a href=\\"https://github.com/iterative/dvc-cml#env-variables\\">setup rev environment variable</a> accordingly </p>
</blockquote>
<h4 id=\\"metrics\\">Metrics</h4>
<p>No metrics available </p>
<h4 id=\\"data\\">Data</h4>
<p>No metrics available </p>
<h4 id=\\"other-experiments\\">Other experiments</h4>
<p>No other experiments available</p>
      </div>
  </body>
</html>
"
`;

exports[`Dvc Report no data Set tag prefix wont show warning 1`] = `
"### Baseline: HEAD~1 ( 1111111 vs 0000000 ) 
 

#### Metrics 

 No metrics available 

#### Data 

No metrics available 

#### Other experiments 
No other experiments available"
`;

exports[`Dvc Report no data Set tag prefix wont show warning 2`] = `
"
<!doctype html>
<html>
	<head>
		<meta charset=\\"utf-8\\">
		<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, minimal-ui\\">
		<title>DVC Report</title>
		<link rel=\\"stylesheet\\" href=\\"report.css\\">
		<style>
			body {
				box-sizing: border-box;
				min-width: 200px;
				max-width: 980px;
				margin: 0 auto;
				padding: 45px;
			}
		</style>
	</head>
	<body>
      <div class=\\"markdown-body\\" id=\\"content\\">
        <h3 id=\\"baseline-head1--1111111-vs-0000000-\\">Baseline: HEAD~1 ( 1111111 vs 0000000 )</h3>
<h4 id=\\"metrics\\">Metrics</h4>
<p>No metrics available </p>
<h4 id=\\"data\\">Data</h4>
<p>No metrics available </p>
<h4 id=\\"other-experiments\\">Other experiments</h4>
<p>No other experiments available</p>
      </div>
  </body>
</html>
"
`;
